src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsassessforearlyreleaseapi/service/OffenderService.kt

```diff
@@ -154,21 +162,23 @@ class OffenderService(
     ),
   )
 
-  fun Assessment.toOffenderSummary() = OffenderSummary(
-    prisonNumber = offender.prisonNumber,
-    bookingId = offender.bookingId,
-    forename = offender.forename!!,
-    surname = offender.surname!!,
-    hdced = offender.hdced,
-    workingDaysToHdced = workingDaysService.workingDaysBefore(offender.hdced),
-    probationPractitioner = this.responsibleCom?.fullName,
-    isPostponed = this.status == AssessmentStatus.POSTPONED,
-    postponementDate = this.postponementDate,
-    postponementReasons = this.postponementReasons.map { reason -> reason.reasonType }.toList(),
-    status = this.status,
-    addressChecksComplete = this.addressChecksComplete,
-    currentTask = this.currentTask(),
-    taskOverdueOn = offender.sentenceStartDate?.plusDays(DAYS_BEFORE_SENTENCE_START),
+  private fun createOffenderSummaryResponse(assessment: Assessment): OffenderSummaryResponse = OffenderSummaryResponse(
```

Comment:
was there a reason you changed this from an extension function? 
We use this pattern all over the place

Reply:
Generally the convention i think is the the service or a mapper/builder should know how to convert something not one of  the objects used to create the offender summary. Extension function are powerful but can be difficult to maintain and can break rules of responsibility.  Also if we used a mapper/builder approach in separate class we can test independently.

Reply:
It's scoped to the service so the responsibility and logic is sat within the service rather than domain entity and the domain entity still doesn't know anything about the api or response objects

Reply:
I get the scope thing,... but why not just add it to a mapper, were every one knows were it should be implemented what does it actually give us in this case and what if some one else add's another extension with same name in a different scope... my be I just don't like extension function unless they are used in a very particular way. Yet's have a chat before we merge!?